examples/ants_continuous.jl

```diff
@@ -0,0 +1,434 @@
+# # Ants 
+
+# This is a implementation of the original ants netlogo model (https://ccl.northwestern.edu/netlogo/models/Ants),
+# with some adaptations to work with a grid similar to the mesa model in (https://github.com/mgoadric/ants-mesa)
+
+# This model describes a colony of ants that wander off from the colony and find food to bring back.
+# The ants move around randomly at first, and when encountering food, return to their colony leaving a pheromone trail
+# Other ants when moving randomly could also follow the pheromone trail in the strongest direction
+
+using Base: String, Float64, Tuple
+using Agents, AgentsPlots
+using GLMakie
+using InteractiveDynamics
+using ImageFiltering: imfilter as matrix_correlation
+using DataFrames: rename!, stack
+using LinearAlgebra: normalize
+using Distributions: Uniform
+
+
+"""
+Agent that represents the colony
+"""
+mutable struct Colony <: AbstractAgent
```

Comment:
I'd say that some of your issues are coming about from mixing agents that sit on `GridSpace` and `ContinuousSpace` in the same model. 

Pretty sure we should actually disallow that in general.

I think the better way to implement this example would be to have Colony and Food as model properties actually. They don't need to be agents. Similar to how Grass is made in [Predator - Prey](https://juliadynamics.github.io/Agents.jl/stable/examples/predator_prey_fast/).

Reply:
Ahah interesting @Libbum , i'm working on the same example for an open source project dedicated to distributed computing (openMole) !!
You will find my work (in progress) here :  https://github.com/reyman/julia-antworld-model

Reply:
I push a much much better version tonight :)
Not perfect, but that's also a start using both env for continuous and grid ... 

https://user-images.githubusercontent.com/466329/124834777-95627e00-df80-11eb-8b69-af9cb23716de.mp4

Reply:
I push an ultimate version, things that miss : 
- [ ] get the front patch (like the patch-right-and-ahead 1, patch-left-and-ahead 1 from netlogo)
- [ ] better random path (linked to 1, i need to select the best pheromone in the front ant patch only)